src/containers/StructurePage/components/ResourceGroup.jsx

```diff
@@ -8,41 +8,100 @@
 
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { Button } from 'ndla-ui';
 import { injectT } from 'ndla-i18n';
-import { ResourceTypeShape } from '../../../shapes';
+import { Plus } from 'ndla-icons/action';
+import BEMHelper from 'react-bem-helper';
+
 import Accordion from '../../../components/Accordion';
 import ResourceItem from './ResourceItem';
+import AddResourceModal from './AddResourceModal';
 
+const classes = new BEMHelper({
+  name: 'topic-resource',
+  prefix: 'c-',
+});
 class ResourceGroup extends PureComponent {
   constructor(props) {
     super(props);
     this.state = {
       displayResource: false,
+      showAddModal: false,
     };
+    this.handleToggle = this.handleToggle.bind(this);
+    this.toggleAddModal = this.toggleAddModal.bind(this);
+  }
+
+  handleToggle() {
+    this.setState(prevState => ({
+      displayResource: !prevState.displayResource,
+    }));
+  }
+
+  toggleAddModal() {
+    this.setState(prevState => ({ showAddModal: !prevState.showAddModal }));
   }
 
   render() {
-    const { icon, topicResource } = this.props;
+    const {
+      icon,
+      resource,
+      topicResource,
+      t,
+      params,
+      refreshResources,
+    } = this.props;
 
     return (
-      <Accordion
-        handleToggle={() =>
-          this.setState(prevState => ({
-            displayResource: !prevState.displayResource,
-          }))
-        }
-        resourceGroup
-        header={topicResource.name}
-        hidden={this.state.displayResource}>
-        <ResourceItem resources={topicResource.resources} {...{ icon }} />
-      </Accordion>
+      <React.Fragment>
+        <Accordion
+          addButton={
+            <Button
+              {...classes('addButton')}
+              stripped
+              onClick={this.toggleAddModal}>
+              <Plus />
+              {t('taxonomy.addResource')}
+            </Button>
+          }
+          handleToggle={this.handleToggle}
+          resourceGroup
+          header={resource.name}
+          hidden={topicResource.resources ? this.state.displayResource : true}>
+          {topicResource.resources && (
+            <ResourceItem
+              resources={topicResource.resources}
+              {...{ icon, classes }}
```

Comment:
`icon={icon}` og classes kan du vel eksporteres og importeres? Da slipper man å sende classes som props langt ned i komponent hierarkiet og man kan bare importere det der man trenger classes.

Reply:
Enig, men fikser det i delete-resource branchen for har refaktorert en del der